### PR TITLE
VideoPress: Add `upload-jwt` endpoint

### PR DESCRIPTION
**Pull request by renatoagds**
Introduce an endpoint and utility class to handle the VideoPress token.

Related with #26316 

#### Changes proposed in this Pull Request:

* Add new class `VideoPressToken`.
* Add new endpoint `upload-jwt`.
* Use `VideoPressToken` static methods to retrieve tokens in other parts.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

_New Class_
* Build the plugin itself (`jetpack build plugins/jetpack`)
* Run `jetpack docker wp shell`
* Execute call for all methods:
* `Automattic\Jetpack\VideoPress\VideoPress_Rest_Api_V1_Token::upload_jwt();`
* `Automattic\Jetpack\VideoPress\VideoPressToken::videopress_playback_jwt();`
* `Automattic\Jetpack\VideoPress\VideoPressToken::videopress_onetime_upload_token();`
* `Automattic\Jetpack\VideoPress\VideoPressToken::videopress_upload_jwt();`

_Edit Video_

* Navigate to VideoPress dashboard.
* Edit poster for some video.
* Everything should work.

### CODE REVIEW

**Comment by retrofox** [comment]
`class-videopresstoken.php filename doesn't see fit with the other files

```
class-wpcom-rest-api-v2-attachment-field-videopress.php
class-wpcom-rest-api-v2-attachment-videopress-data.php
class-wpcom-rest-api-v2-endpoint-videopress.php
```

maybe we should rename it to `class-videoress-rest-api-v1-jwt-endpoint.php` to something like that?

File: projects/packages/videopress/src/class-videopresstoken.php [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+use Jetpack_Options;
+use WP_REST_Response;
+
+/**
+ * VideoPress token utility class
+ */
+class VideoPressToken {
```
Comment (retrofox) [inline projects/packages/videopress/src/class-videopresstoken.php:16]: I think the class name should follow the convention, something like `VIDEOPRESS_REST_API_V1_...`

**Review by retrofox — COMMENTED** [review]
I'm not familiar with the context, so testing gets a little bit tricky to me.

I was able to run the `Automattic\Jetpack\VideoPress\VideoPressToken::upload_jwt();` method, getting the token, URL and blog_id.

The other methods trigger errors probably because I don't know how or what to test :-/

I confirm the PR also registers the new `videopress/v1` namespace/version.

* Navigate to VideoPress dashboard.
* Edit poster for some video.
* Everything should work.

⬆️ works as expected.

Should we ask Kernel folks for help? If I'm not wrong, these changes will impact stuff in production.

**Comment by renatoagds** [comment]
> maybe we should rename it to class-videoress-rest-api-v1-jwt-endpoint.php to something like that?

maybe, but at the current moment we have another class inserting `videopress/v1` endpoint called `class-uploader-rest-endpoints`, so, I think it is the case to define a pattern for this.

> Should we ask Kernel folks for help? If I'm not wrong, these changes will impact stuff in production.

sure. I'll ping them

File: projects/packages/videopress/src/class-videopresstoken.php (hunk 1 above)
Comment (sergeymitr) [reply projects/packages/videopress/src/class-videopresstoken.php:16]: I agree with @retrofox.
The class is not a simple token handler (e.g. `Automattic\Jetpack\Connection\Tokens`), but actually a REST endpoint, so we should name it accordingly.

File: projects/packages/videopress/src/class-videopresstoken.php [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+	 *
+	 * @return void
+	 */
+	public static function register_rest_endpoints() {
+		register_rest_route(
+			'videopress/v1',
+			'upload-jwt',
+			array(
+				'methods'             => \WP_REST_Server::EDITABLE,
+				'callback'            => __CLASS__ . '::upload_jwt',
```
Comment (sergeymitr) [inline projects/packages/videopress/src/class-videopresstoken.php]: Just nitpicking, how do you feel about replacing `__CLASS__` in callbacks with `static::class` for late static binding?

Same goes for all other `self::`, replacing them with `static::` might help a lot in the future.

**Review by sergeymitr — DISMISSED** [review]
Looks good, just a couple of minor notes.

File: projects/packages/videopress/src/class-videopresstoken.php (hunk 1 above)
Comment (renatoagds) [reply projects/packages/videopress/src/class-videopresstoken.php:16]: Done [9be629b](https://github.com/Automattic/jetpack/pull/26406/commits/9be629b56940b8c46ad3e07753ecfcb10792a63c)

File: projects/packages/videopress/src/class-videopresstoken.php (hunk 2 above)
Comment (renatoagds) [reply projects/packages/videopress/src/class-videopresstoken.php]: Done: [4c1b949](https://github.com/Automattic/jetpack/pull/26406/commits/4c1b949455d49a620cd985b720f4f58a14c24def)

**Review by kraftbj — APPROVED** [review]

**Pull request merged**